pkg/manager/leaderelection.go

```diff
@@ -13,13 +14,18 @@ func (k *KubeMacPoolManager) waitToStartLeading() error {
 }
 
 func (k *KubeMacPoolManager) markPodAsLeader() error {
-	pod, err := k.clientset.CoreV1().Pods(k.podNamespace).Get(context.TODO(), k.podName, metav1.GetOptions{})
-	if err != nil {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Can you please elaborate why do we need RetryOnConflict?

Reply:
If k8s changes something like annotation at pods (there is a lot of stuff going on at annotations and labels that we don't control) this is going to conflict, better be safe here, it does not introduce any cost.